community/detectors/apache_airflow_cve_2020_17526/src/test/java/com/google/tsunami/plugins/cve202017526/Cve202017526DetectorTest.java

```diff
@@ -0,0 +1,220 @@
+                .injectMembers(this);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        mockTargetService.shutdown();
+        mockCallbackServer.shutdown();
+    }
+
+    @Test
+    public void detect_withCallbackServer_onVulnerableTarget_returnsVulnerability()
+            throws IOException {
+        startMockWebServer();
+        mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());
+        NetworkService targetNetworkService =
+                NetworkService.newBuilder()
+                        .setNetworkEndpoint(
+                                forHostnameAndPort(mockTargetService.getHostName(), mockTargetService.getPort()))
+                        .addSupportedHttpMethods("POST")
+                        .build();
+        TargetInfo targetInfo =
+                TargetInfo.newBuilder()
+                        .addNetworkEndpoints(targetNetworkService.getNetworkEndpoint())
+                        .build();
+
+        DetectionReportList detectionReports =
+                detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+
+        Truth.assertThat(mockCallbackServer.getRequestCount()).isEqualTo(1);
+        assertThat(detectionReports.getDetectionReportsList())
+                .comparingExpectedFieldsOnly()
+                .containsExactly(
+                        DetectionReport.newBuilder()
+                                .setTargetInfo(targetInfo)
+                                .setNetworkService(targetNetworkService)
+                                .setDetectionTimestamp(
+                                        Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                                .setVulnerability(
+                                        Vulnerability.newBuilder()
+                                                .setMainId(
+                                                        VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                                                                .setValue("CVE-2020-17526"))
+                                                .setSeverity(Severity.CRITICAL)
+                                                .setTitle("CVE-2020-17526 Authentication bypass lead to Arbitrary Code Execution in Apache Airflow")
+                                                .setDescription(
+                                                        "An attacker can use a default DAG to execute arbitrary code on"
+                                                                + " the server hosting the apache airflow application.")
+                                                .setRecommendation(
+                                                        "update to higher versions or Change default value for [webserver] secret_key config"))
```

Comment:
Please consider providing recommendations regarding an upgrade to a specific non-vulnerable version of Apache Airflow